plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/index.tsx

```diff
@@ -31,14 +32,25 @@ const Edit = ( props: ProductCollectionEditComponentProps ) => {
 		[ clientId ]
 	);
 
-	const productCollectionUIStateInEditor =
-		getProductCollectionUIStateInEditor( {
-			hasInnerBlocks,
+	const { productCollectionUIStateInEditor, isLoading } =
+		useProductCollectionUIState( {
 			location,
-			attributes: props.attributes,
-			usesReference: props.usesReference,
+			attributes,
+			hasInnerBlocks,
+			...( props.usesReference && {
+				usesReference: props.usesReference,
+			} ),
```

Comment:
What's the benefit of not passing `usesReference` if it's not available? Because I think `usesReference: props.usesReference,` is cleaner and more explicit, even if sometimes value will be `undefined`.

Reply:
I bet it's just a TS thing 😅

Reply:
Yeah, It was TS thing but I was able to simplify it in [7380aa8](https://github.com/woocommerce/woocommerce/pull/51114/commits/7380aa818e946ca0f802ee1d799a92c4dc4b94b9) 😀